assets/wizards/setup/views/settings/index.js

```diff
@@ -8,48 +8,134 @@ import apiFetch from '@wordpress/api-fetch';
 /**
  * Internal dependencies
  */
-import { SelectControl, withWizardScreen, hooks } from '../../../../components/src';
+import {
+	Card,
+	Grid,
+	ImageUpload,
+	SectionHeader,
+	SelectControl,
+	TextControl,
+	withWizardScreen,
+	hooks,
+} from '../../../../components/src';
+
+const pageTitleTemplate = document.title.replace( newspack_aux_data.site_title, '__SITE_TITLE__' );
 
 /**
  * Settings Setup Screen.
  */
-const Settings = ( { setError, wizardApiFetch } ) => {
-	const updateProfile = profile =>
-		apiFetch( {
-			path: '/newspack/v1/profile/',
-			method: 'POST',
-			data: { profile },
-		} );
-	const [ { currencies = [], countries = [] }, setOptions ] = useState( {} );
-	const [ profileData, updateProfileData, isLoading ] = hooks.useObjectState( {}, updateProfile );
+const Settings = ( { setError, wizardApiFetch, renderPrimaryButton } ) => {
+	const [ { currencies = [], countries = [], wpseoFields = [] }, setOptions ] = useState( {} );
+	const [ profileData, updateProfileData, isLoading ] = hooks.useObjectState( {} );
 
 	useEffect( () => {
 		wizardApiFetch( { path: '/newspack/v1/profile/', method: 'GET' } )
 			.then( response => {
-				setOptions( { currencies: response.currencies, countries: response.countries } );
+				setOptions( {
+					currencies: response.currencies,
+					countries: response.countries,
+					wpseoFields: response.wpseo_fields,
+				} );
 				updateProfileData( response.profile, { skipCallback: true } );
 			} )
 			.catch( setError );
 	}, [] );
 
+	const updateProfile = () =>
+		apiFetch( {
+			path: '/newspack/v1/profile/',
+			method: 'POST',
+			data: { profile: profileData },
+		} );
+
+	useEffect( () => {
+		if ( typeof profileData.site_title === 'string' ) {
+			document.title = pageTitleTemplate.replace( '__SITE_TITLE__', profileData.site_title );
+		}
+	}, [ profileData.site_title ] );
+
+	const renderSetting = ( { options, label, key, type, placeholder, className } ) => {
```

Comment:
Nit: why not make this a React component?

Reply:
Hm, it sort of is and isn't a real component - it's more of a switch statement lifted to a function. I'd keep it as it is.